react/src/components/DateInput/index.js

```diff
@@ -9,48 +9,61 @@ class DateInput extends React.PureComponent {
     this.inputRef = React.createRef()
   }
 
-  onChange = () => {
-    const that = this
-    setTimeout(() => {
-      let val = that.inputRef.current.dataset.date
-      if (val === '' || val === 'null') val = null
-      that.props.onChange(val)
-    }, 0)
+  componentDidMount () {
+    if (this.unbindInput) this.unbindInput()
+
+    this.inputRef.current.addEventListener('change', this.onChange)
+
+    this.unbindInput = () => {
+      this.inputRef.current.removeEventListener('change', this.onChange)
+      delete this.unbindInput
+    }
   }
 
-  focus () {
-    this.inputRef.current.focus()
+  componentWillUnmount () {
+    if (this.unbindInput) this.unbindInput()
   }
 
-  componentDidMount () {
-    if (this.props.type === 'date' && window.folioConsoleInitDatePicker) {
-      window.folioConsoleInitDatePicker(this.inputRef.current)
-    } else if (this.props.type === 'datetime' && window.folioConsoleInitDateTimePicker) {
-      window.folioConsoleInitDateTimePicker(this.inputRef.current)
-    }
+  formatDate (date, delimiter = '-') {
+    const options = { day: "2-digit", month: "2-digit", year: "numeric"}
+    const dateObj = new Date(date)
 
-    window.jQuery(this.inputRef.current).on('dp.change', this.onChange)
+    return dateObj.toLocaleDateString("cs-CZ", options)
   }
 
-  componentWillUnmount () {
-    window.jQuery(this.inputRef.current).off('dp.change', this.onChange)
-    if (window.folioConsoleUnbindDatePicker) {
-      window.folioConsoleUnbindDatePicker(this.inputRef.current)
-    }
+  onChange = (event) => {
```

Comment:
@VladaTrefil neni tohle zbytecne? Melo by stacit `onChange={this.props.onChange}` z inputu, ne?

Reply:
@mreq Pisu o tom v issue, melo by to byt zbytecne, ale neni, protoze musim value naformatovat pro ulozeni, misto toho abych poslal rovnou value, nebo idealne Date objekt.

https://github.com/sinfin/folio/issues/269#issuecomment-2165218641

Reply:
Byl v tom trochu gulas. Sjednotil jsem praci s datumy obecne pro atomy. React cast pak uz bylo snadne doladit. Diky :+1: